lib/pages/dashboard/dashoard_page.dart

```diff
@@ -22,17 +38,53 @@ class _DashboardPageState extends State<DashboardPage> {
       sliver: SliverList(
         delegate: SliverChildListDelegate.fixed([
           Paragraph(title: "Vos clubs"),
-          DashboardClubs(onClubChange: (club) => _selectCurrentClub(club)),
+          DashboardClubs(key: ValueKey("dashboard-clubs"), onClubChange: (club) => _selectCurrentClub(club)),
           Paragraph(title: "Vos équipes"),
-          if (_currentClub != null) DashboardClubTeams(club: _currentClub, onTeamFavoriteChange: (_) => _selectCurrentClub(_currentClub)),
+          if (_currentClub != null) _buildDashboardElement(_teamsElement),
           Paragraph(title: "Votre agenda"),
-          DashboardAgenda()
+          if (_currentClub != null) _buildDashboardElement(_agendaElement)
         ])
       )
     );
   }
 
+  Widget _teamsElement(List<Team> teams) {
+    if (teams.length > 0) {
+      return DashboardClubTeams(teams: teams, onFavoriteTeamsChange: () => _selectCurrentClub(_currentClub));
+    } else {
+      return Center(
+        child: Padding(
+          padding: const EdgeInsets.all(28.0),
+          child: RaisedButton(
+            onPressed: () =>  Router.push(context: context, builder: (_) => ClubDetailPage(_currentClub)).then(
+              (_) => _selectCurrentClub(_currentClub),
+            ),
+            padding: EdgeInsets.all(12.0),
+            child: Text("Sélectionnez une équipe favorite")
+          ),
+        ),
+      );
+    }
+  }
+
+  Widget _agendaElement(List<Team> teams) {
+    return DashboardAgenda(teams: teams);
+  }
+
+  Widget _buildDashboardElement(Function toBuild) {
```

Comment:
The main difference between Javascript and Dart is that Dart is strong typed language so instead of using `Function` (which can't define parameters type), can't you define a `typedef`? 
*BTW: a builder function often needs the `BuildContext`*
```suggestion
typedef DashboardElementBuilder = Function(BuildContext, List<Team>)
Widget _buildDashboardElement(DashboardElementBuilder builder) {
  ...
}  
```
Then you should be able to use `_agendaElement` and other methods as they have the same signature (to be verified)